File: src/main/java/com/geopark/web/model/entity/Earthquake.java
package com.geopark.web.model.entity;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;

/**
 * <p>
 * 地灾信息
 * </p>
 *
 * @author lichee
 * @since 2019-08-14
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("earthquake")
public class Earthquake implements Serializable {

    @ExcelIgnore
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    @ExcelIgnore
    private Integer id;

    /**
     * 地质公园ID
     */
    @ExcelProperty(value = "地质公园ID",index = 0)
    private Integer parkId;

    /**
     * 灾害点编号
     */
    @ExcelProperty(value = "灾害点编号",index = 1)
    private String disasternumber;

    /**
     * 灾害点名称
     */
    @ExcelProperty(value = "灾害点名称",index = 2)
    private String disastername;

    /**
     * 类型
     */
    @ExcelProperty(value = "类型",index = 3)
    private String type;

    /**
     * 地理位置
     */
    @ExcelProperty(value = "地理位置",index = 4)
    private String position;

    /**
     * 坐标
     */
    @ExcelProperty(value = "坐标",index = 5)
    private String coordinate;

    /**
     * 规模
     */
    @ExcelProperty(value = "规模",index = 6)
    private String scale;

    /**
     * 稳定性
     */
    @ExcelProperty(value = "稳定性",index = 7)
    private String stability;

    /**
     * 危险性
     */
    @ExcelProperty(value = "危险性",index = 8)
    private String dangerous;

    /**
     * 特征描述
     */
    @ExcelProperty(value = "特征描述",index = 9)
    private String description;

    /**
     * 诱发因素
     */
    @ExcelProperty(value = "诱发因素",index = 10)
    private String factor;

    /**
     * 威胁对象
     */
    @ExcelProperty(value = "威胁对象",index = 11)
    private String threat;

    /**
     * 备注
     */
    @ExcelProperty(value = "备注",index = 12)
    private String note;

}

File: src/main/resources/static/js/pkmanage/earthquake.js
//公园概况

layui.use(['config', 'lichee', 'jquery', 'layer', 'table', 'form', 'laydate'], function () {

    var config = layui.config;
    var lichee = layui.lichee;
    var $ = layui.jquery;
    var layer = layui.layer;
    var table = layui.table;
    var form = layui.form;
    var laydate = layui.laydate;

    var tableInfo = table.render({
        elem: '#table',
        toolbar: '#toolbar',
        defaultToolbar: [],
        //width:'auto',
        url: '/earthquake/page',
        title: '地灾信息',
        page: true,
        headers: {Authorization: config.getToken()},
        request: config.request,
        parseData: config.parseData,
        response: config.response,
        cols: [[
            //{type: 'checkbox', fixed: 'left'},
            // {field: 'id', title: 'ID'},
            {field: 'parkId', align: 'center', sort: true, title: '地质公园ID'},
            {field: 'disasternumber', align: 'center', sort: true, title: '灾害点编号'},
            {field: 'disastername', align: 'center', sort: true, title: '灾害点名称'},
            {field: 'type', align: 'center', sort: true, title: '类型'},
            {field: 'position', align: 'center', sort: true, title: '地理位置'},
            {field: 'coordinate', align: 'center', sort: true, title: '坐标'},
            {field: 'scale', align: 'center', sort: true, title: '规模'},
            {field: 'stability', align: 'center', sort: true, title: '稳定性'},
            {field: 'dangerous', align: 'center', sort: true, title: '危险性'},
            //{field: 'description', align: 'center', sort: true, title: '特征描述'},
            //{field: 'factor', align: 'center', sort: true, title: '诱发因素'},
            //{field: 'threat', align: 'center', sort: true, title: '威胁对象'},
            //{field: 'note', align: 'center', sort: true, title: '备注'},
            {fixed: 'right',align : 'center', title: '操作', toolbar: '#bar', width: 180}
        ]]
    });

    table.on('toolbar(tableFilter)', function (obj) {
        var checkStatus = table.checkStatus(obj.config.id);
        switch (obj.event) {
            case 'add':
                showEditModel();
                break;
            case 'getCheckLength':
                var data = checkStatus.data;
                layer.msg('选中了：' + data.length + ' 个');
                break;
            case 'isAll':
                layer.msg(checkStatus.isAll ? '全选' : '未全选');
                break;
        };
    });

    table.on('tool(tableFilter)', function (obj) {
        var data = obj.data;
        if (obj.event === 'del') {
            layer.confirm('确定要删除吗？', function (index) {
                layer.load(2);
                lichee.delete('/earthquake/' + obj.data.id, {}, function () {
                    layer.closeAll('loading');
                    layer.msg('删除成功', {icon: 1});
                    obj.del();
                });
            });
        } else if (obj.event === 'edit') {
            showEditModel(data);
        }
    });

    var showEditModel = function (data) {
        layer.open({
            title: data ? '详情/修改' : '添加',
            type: 1,
            area: '800px',
            offset: '20px',
            content: $('#form-model').html(),
            success: function () {
                $('#form')[0].reset();
                lichee.get('/park/list', {async: false}, function (data) {
                    $('#parkId').vm({parks: data.result});
                    form.render('select');
                });
                if (data) {
                    form.val('formFilter', data);
                }
                $('#form .close').click(function () {
                    layer.closeAll('page');
                });
            }
        });
    };

    var form = layui.form;
    form.on('submit(formSubmit)', function (data) {
        layer.load(2);
        if (data.field.id) {
            lichee.put('/earthquake/' + data.field.id, {data: data.field}, function (res) {
                callFunction(res);
            });
        } else {
            lichee.post('/earthquake', {data: data.field}, function (res) {
                callFunction(res);
            });
        }
        return false;
    });

    form.verify({
        commonLength: [/^[\S]{1,100}$/, '字符长度最大100'],
        descLength: [/^[\S]{0,180}$/, '字符长度最大180']
    });

    var callFunction = function (res) {
        layer.closeAll();
        if (res.status == 200) {
            layer.msg("修改成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        } else if(res.status == 201)  {
            layer.msg("添加成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        }
    };

    $('#searchButton').click(function () {
        tableInfo.reload({where: lichee.getSearchForm()});
    });

    laydate.render({
        elem: '#startTime',
        type: 'datetime',
        value: '',
        trigger: 'click'
    });

    $(".laydate").each(function (i,e) {
        laydate.render({
            elem: this,
            type: 'datetime',
            value: '',
            trigger: 'click'
        });
    });

});